docs/guidelines/git-github.md

```diff
@@ -0,0 +1,306 @@
+
+- Verify that Github accepts the key, test the connection
+
+(When testing the connection to github.com, ssh will ask if the fingerprint of the SSH key presented by github.com is ok and if you would like to continue. If you are connecting to github.com answer "yes". Understanding this trust chain is not for this guide - but if you want to validate the fingerprint that's suggested you can correlate it to the [official ssh key fingerprints](https://docs.github.com/en/authentication/keeping-your-account-and-data-secure/githubs-ssh-key-fingerprints). The known hosts file that is mentioned is the `~/.ssh/known_hosts` file)
+
+```shell
+ssh -T git@github.com
+```
+
+The command should a message similar to this to indicate success:
+
+```
+Hi larskaare! You've successfully authenticated, but GitHub does not provide shell access.
+```
+
+### Using Git and SSH with github
+
+A this stage we have SSH all configured on both ends. However, how do we tell git to use SSH? You typically would do this when cloning a repo or configuring the `remote`
+
+- A git clone command like this `git clone git@github.com:equinor/appsec.git` tells git to use SSH. The same command for using HTTPS would be git `git clone https://github.com/equinor/appsec.git`. The giveaway for HTTPS would be the url which starts with https 😅 That is not included when we use SSH - we rather have the `git@github.com` part of the url.
+- You can examine the remote of a repo to identify if it uses SSH or HTTPS
+
+```shell
+git remote -v
+```
+
+should out output remote name and a URI containing the `git@github.com`
+
+```
+origin	git@github.com:equinor/appsec.git (fetch)
+origin	git@github.com:equinor/appsec.git (push)
+```
+
+- You can add and change remotes for a cloned repo, from HTTPS to SSH using the command `git remote set-url <remote name> <new uri>`
+
+### Configure the SSH key for usage with Equinor's SSO protected resources
+
+The Equinor organization on github.com is protected behind SSO login. In order for your SSH key to be used with resources in the Github "Equinor" or "Equinor-Playground" organizations you need to authorize the key for these permissions on your behalf. [Github documentation](https://docs.github.com/en/enterprise-cloud@latest/authentication/authenticating-with-saml-single-sign-on/authorizing-an-ssh-key-for-use-with-saml-single-sign-on) gives you all the details.
+
+- To authorize your key head over to your [SSH Key Settings](https://github.com/settings/keys) on your profile on Github, select the key in question, select "Configure SSO" and authorize against organisation(s).
+
+### Rotating SSH keys
+
+!!! Tip "Rotate your SSH keys"
+
+    Your SSH keys, and passphrases, should be rotated at least on a yearly basis. Put an re-occurring appointment in your calender for this. This process could be automated, but doing it once in a while in manual mode may help you not to forget how things work.
```

Comment:
Is there not governing documents that state this too?
Should we reference it?

Reply:
There used to be. I'll look into it and update if later if needed.